apps/webstore/src/app/pages/auth/checkout/checkout.component.ts

```diff
@@ -236,13 +249,89 @@ export class CheckoutComponent implements OnInit {
   }
 
   submitCheckout() {
+    // check checkout form and if terms are accepted
+    this.checkFormValidity();
+
     if (this.isLoggedIn) {
       this.createOrder();
     } else {
       this.createOrderWithNewUser();
     }
   }
 
+  checkFormValidity() {
+    if (!this.checkoutForm.valid || !this.isTermsAndConditionsAccepted) {
+      this.checkoutInvalid = true;
+      this.checkShippingInputFields();
+      if (!this.billingAddressIsTheSame) {
+        if (!this.billingAddressForm.valid) {
+          this.billingAddressInvalid = true;
+          this.checkBillingInputFields();
+          return;
+        }
+      }
+      return;
+    }
+
+    // check if billing address is the same
+    if (!this.billingAddressIsTheSame) {
+      if (!this.billingAddressForm.valid) {
+        this.billingAddressInvalid = true;
+        this.checkBillingInputFields();
+        return;
+      }
+    }
+
+    this.checkoutInvalid = false;
+  }
+
+  checkShippingInputFields() {
```

Comment:
Multiple methods are missing return types
```suggestion
  checkShippingInputFields(): void {
```